client/client.go

```diff
@@ -793,23 +793,20 @@ tsoBatchLoop:
 			if stream == nil {
 				log.Info("[pd] tso stream is not ready", zap.String("dc", dc))
 				c.updateConnectionCtxs(dispatcherCtx, dc, &connectionCtxs)
-				if retryTimeConsuming >= c.option.timeout {
+				select {
+				case <-dispatcherCtx.Done():
+					return
+				case <-streamLoopTimer.C:
+					streamLoopTimer.Reset(c.option.timeout)
 					err = errs.ErrClientCreateTSOStream.FastGenByArgs("retry timeout")
 					log.Error("[pd] create tso stream error", zap.String("dc-location", dc), errs.ZapError(err))
 					c.ScheduleCheckLeader()
 					c.finishTSORequest(tbc.getCollectedRequests(), 0, 0, 0, errors.WithStack(err))
-					retryTimeConsuming = 0
 					continue tsoBatchLoop
-				}
-				select {
-				case <-dispatcherCtx.Done():
-					return
 				case <-time.After(time.Second):
-					retryTimeConsuming += time.Second
-					continue
+					continue streamChoosingLoop
 				}
 			}
-			retryTimeConsuming = 0
```

Comment:
Will the new implementation timeout imediately when back to streamChoosingLoop?

Reply:
No, it will check whether stream is created. If created, it will not timeout